### PR TITLE
Treat Property Let, Get, and Set equally.

### LINKED ISSUE #3340: Code Explorer sorting by type should keep property accessors together
Currently `PropertyGet` accessors show up before `PropertyLet` accessors. That's nice, except if a property has both accessors, then navigating properties becomes ...annoying:

image

ByType-sort should keep property accessors bundled together, so `PropertyGet` and `PropertyLet` and `PropertySet` should have the same weight in sorting, and properties need to be special-cased so that instead of this:

 - CategoryCode (Get)
 - 20 other members
 - CategoryCode (Let)

I'd have this:

 - CategoryCode (Get)
 - CategoryCode (Let)
 - 20 other members

### PR DESCRIPTION
**Pull request by shadowofsilicon**
This should group them together and close #3340

### CODE REVIEW

File: RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerItemViewModel.cs [hunk 1]
```diff
@@ -31,7 +31,8 @@ public class CompareByType : Comparer<CodeExplorerItemViewModel>
     {
         private static readonly Dictionary<DeclarationType, int> SortOrder = new Dictionary<DeclarationType, int>
         {
-            // Some DeclarationTypes we want to treat the same, like Subs and Functions.
+            // Some DeclarationTypes we want to treat the same, like Subs and Functions,
```
Comment (retailcoder) [inline RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerItemViewModel.cs]: I'm not sure `Sub` and `Function` should be treated the same, but depending on sorting options... is there a way to have the `Public` members should show up before the `Private` ones?

File: RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerItemViewModel.cs [hunk 2]
```diff
@@ -31,7 +31,8 @@ public class CompareByType : Comparer<CodeExplorerItemViewModel>
     {
```
Comment (retailcoder) [inline RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerItemViewModel.cs]: Does `NameWithSignature` include the access modifier for e.g. an `Implicit` public procedure? i.e. doesn't `Public Sub Zoo()` come before `Sub Foo()` if that's the case?

**Comment by shadowofsilicon** [comment]
Sorting Public above Private shouldn't be difficult at all.  I'll change that.

I have no problem sorting Subs and Functions separately, but then we need two new icons (one for Public and one for Private.)

**Comment by retailcoder** [comment]
We're supposed to have objectmethodprivate for private methods, and 
objectmethod for public ones...

**Comment by shadowofsilicon** [comment]
Is there someplace (another project, website, etc) that we are getting these icons from?

**Comment by retailcoder** [comment]
https://github.com/rubberduck-vba/Rubberduck/tree/next/RetailCoder.VBE/Resources/Custom

**Comment by shadowofsilicon** [comment]
I know where to find the resources in our project.

I see that they exist in our project.  That's good.  I was asking, where did we get those icons from, outside of *our* project?

**Comment by retailcoder** [comment]
They're the SharpDevelop icons, or custom-made from SharpDevelop icon parts (per attributions.md).

**Comment by shadowofsilicon** [comment]
Yes, we have and already use the icons @retailcoder posted above.  They are all in use in the CE.

The proposal is to, inside the CE's TreeView, to sort `Sub`s (`DeclarationType.Procedure`) and `Function`s (`DeclarationType.Function`) separately.

We need a new icon for one of those `DeclarationType`s.  By that, I mean a new icon that we currently do *not* have in RD.

We currently have:
 - objectmethod for `Public` `Sub`s and `Function`s
 - objectmethodprivate for `Private` `Sub`s and `Function`s
- objectmethodfriend for `Friend` `Sub`s and `Function`s

Here are our options from SharpDevelop (in no particular order):

- icons 16x16 extensionmethod
- icons 16x16 delegate
- virtualmethod
- pinvokemethod
- operator
- extensionmethod

I propose using pinvokemethod for `Public` `Function`s, and adding the padlock and the... whatever that blue thing is... to the same image to create icons for `Private` `Function`s and `Friend` `Function`s.  Then the icons currently in use ( objectmethod objectmethodprivate objectmethodfriend ) will be used only for the respective types of `Sub`s.

**Comment by retailcoder** [comment]
That would be misuse of the icons' purpose. Icons have a specific name and a specific purpose. 
image is a public method, and anything with a padlock implies `Private`; the blue cube 
image indicates a `Friend` accessibility.

image looks like a "Run" command icon, or something like that. I don't think we need different icons for `Sub` and `Function` procedure.

The Object Browser doesn't use different icons for `Sub` and `Function` procedure, and no version or edition of Visual Studio ever did either.

image

**Comment by retailcoder** [comment]
What I mean is, icons have a name, and a purpose; they're meant to be used accordingly, not in whichever way we think could possibly make sense, e.g. if image is "implement member", then it should be used on a "implement member" command.

**Comment by shadowofsilicon** [comment]
The Object Browser only organizes by Name in alphabetical order.  It does not group by Type.
image

The CE would behave the same way if you chose "By Name" in the Sort menu, and un-selected "By Type" (which needs to be renamed "Group by Type".)

What we are discussing only matters of "by Type" is selected.  It has to be, that's the only reason we would care what the Type is.

Without a unique icon for `Sub`s and `Function`s, if the user selected "By Name" *and* "By Type" as sort options, this is what they would end up with, assuming they are all `Public`:

image InchesToPoints
image InputBox
image Intersect
image MacroOptions
image OnKey
image OnTime
image RegisterXLL
image Help
image MailLogoff
image MailLogon
image NextLetter
image OnRepeat
image OnUndo
image Quit
image RecordMacro

See how the alphabetical order starts over between `Register XLL` and `Help`?  That's because `Help` and everything after is a `Sub` and `Register XLL` and everything above it is a `Function`, and it will look **broken** if we sort `Sub`s and `Function`s separately.

Are you **sure** you want to group `Subs` and `Functions` separately, with the optional "By Type" is selected?

**Comment by retailcoder** [comment]
Is this whole discussion happening just because of this?

> I'm not sure `Sub` and `Function` should be treated the same, [...]

If it looks weird, then we're done; `Sub` and `Function` are to be treated the same, period 😉 - I was simply voicing that *perhaps maybe* they could be, and as your list of members shows, indeed, it looks clumsy. So..

> Are you sure you want to group Subs and Functions separately, with the optional "By Type" is selected?

Nope.

**Pull request merged**